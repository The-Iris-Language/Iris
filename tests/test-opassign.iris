..tests op assign on ints and floats
class Main () {
    public:
        int univ main() {
            int a = 0;
            a += 2;
            if (a == 2) {
                Olympus.println("successfully added to a");
            }
            a *= 3;
            if (a == 6) {
                Olympus.println("successfully multiplied a"); 
            }

            a -= 1;
            if (a == 5) {
                Olympus.println("successfully subtracted a"); 
            }

            a /= 2;
            if (a == 2) {
                Olympus.println("successfully divided a"); 
            }


            float f = 4.55;
            f += 1.0; 
            if (f == 5.55) {
                Olympus.println("successfully added to f");
            }

            f *= 2.0;
            if (f == 11.1) {
                Olympus.println("successfully multiplied f");
            }

            f -= 1.1;
            if (f == 10.0) {
                Olympus.println("successfully subtracted f");
            }

            f /= 2.0;
            if (f == 5.0) {
                Olympus.println("successfully divided f");
            }
        }
}