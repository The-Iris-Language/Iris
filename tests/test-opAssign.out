int main()
{
int i;
i = 1;
i+=1;
i-=1;
i*=4;
i/=2;
return 0;
}